devices/src/legacy/i8042.rs

```diff
@@ -6,47 +6,314 @@
 // found in the THIRD-PARTY file.
 
 use logger::{Metric, METRICS};
+use std::fmt;
+use std::num::Wrapping;
 use std::{io, result};
 use sys_util::EventFd;
 
 use BusDevice;
 
-const RESET_CMD: u8 = 0xfe;
+#[derive(Debug)]
+pub enum Error {
+    CloneCpuResetEvt(io::Error),
+    KbdInterruptDisabled,
+    KbdInterruptFailure(io::Error),
+    InternalBufferFull,
+}
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::CloneCpuResetEvt(io_err) => {
+                write!(f, "Could not clone CPU reset eventfd: {:?}.", io_err)
+            }
+            Error::KbdInterruptDisabled => {
+                write!(f, "Keyboard interrupt disabled by guest driver.",)
+            }
+            Error::KbdInterruptFailure(io_err) => {
+                write!(f, "Could not trigger keyboard interrupt: {:?}.", io_err)
```

Comment:
Same for this one (use to_string instead of debug).

Reply:
Done.